src/commonTest/kotlin/UuidTest.kt

```diff
@@ -51,4 +51,31 @@ class UuidTest {
         assertEquals(4, uuid.version)
         assertEquals(2, uuid.variant)
     }
+
+    @Test fun variants() {
+        assertEquals(0, Uuid.parse("00000000-0000-0000-0000-000000000000")!!.variant, "Nil or NCS")
+        assertEquals(2, Uuid.parse("00000000-0000-0000-8000-000000000000")!!.variant, "RFC 4122")
+        assertEquals(6, Uuid.parse("00000000-0000-0000-c000-000000000000")!!.variant, "Microsoft")
+        assertEquals(7, Uuid.parse("00000000-0000-0000-e000-000000000000")!!.variant, "Future")
+    }
+
+    @Test fun versions() {
+        assertEquals(0, Uuid.parse("00000000-0000-0000-0000-000000000000")!!.version, "Nil")
+        assertEquals(1, Uuid.parse("00000000-0000-1000-0000-000000000000")!!.version, "time-based")
+        assertEquals(2, Uuid.parse("00000000-0000-2000-0000-000000000000")!!.version, "DCE security")
+        assertEquals(3, Uuid.parse("00000000-0000-3000-0000-000000000000")!!.version, "name-based using MD5 hashing")
+        assertEquals(4, Uuid.parse("00000000-0000-4000-0000-000000000000")!!.version, "random or pseudo-random")
+        assertEquals(5, Uuid.parse("00000000-0000-5000-0000-000000000000")!!.version, "name-based using SHA-1 hashing")
+
+        assertEquals(6, Uuid.parse("00000000-0000-6000-0000-000000000000")!!.version, "future #1")
+        assertEquals(7, Uuid.parse("00000000-0000-7000-0000-000000000000")!!.version, "future #2")
+        assertEquals(8, Uuid.parse("00000000-0000-8000-0000-000000000000")!!.version, "future #3")
+        assertEquals(9, Uuid.parse("00000000-0000-9000-0000-000000000000")!!.version, "future #4")
+        assertEquals(10, Uuid.parse("00000000-0000-a000-0000-000000000000")!!.version, "future #5")
+        assertEquals(11, Uuid.parse("00000000-0000-b000-0000-000000000000")!!.version, "future #6")
+        assertEquals(12, Uuid.parse("00000000-0000-c000-0000-000000000000")!!.version, "future #7")
+        assertEquals(13, Uuid.parse("00000000-0000-d000-0000-000000000000")!!.version, "future #8")
+        assertEquals(14, Uuid.parse("00000000-0000-e000-0000-000000000000")!!.version, "future #9")
+        assertEquals(15, Uuid.parse("00000000-0000-f000-0000-000000000000")!!.version, "future #10")
```

Comment:
Here you can see what happens if a function that works in 99% ends up returning `null`. `null` should be used if it is a valid value and indicates absence but not for errors.